src/Covenant/ASG.hs

```diff
@@ -127,41 +153,35 @@ data Scope (args :: Natural) (lets :: Natural) = Scope
 --
 -- @since 1.0.0
 emptyScope :: Scope 0 0
-emptyScope = Scope
+emptyScope = Scope Vector.empty Vector.empty
 
 -- | Construct a literal (constant) value.
 --
 -- @since 1.0.0
-lit ::
-  forall (m :: Type -> Type).
-  (MonadHashCons Id ASGNode m) =>
-  AConstant -> m Id
-lit = refTo . LitInternal
+lit :: AConstant -> ASGCompiler Id
+lit c = do
+  ty <- liftTypeError (typeLit c)
+  refTo (LitInternal ty c)
```

Comment:
I've had to make this a concrete monad, to insert an ExceptT into the stack.

I felt it wouldn't be right to insert ExceptT into HashConsT.
So I defined a new stack ASGCompiler.